app/src/main/java/com/android/voyageur/ui/trip/AddActivity.kt

```diff
@@ -84,12 +79,27 @@ fun AddActivityScreen(tripsViewModel: TripsViewModel, navigationActions: Navigat
 
     val today = normalizeToMidnight(Date())
     val dateNormalized = activityDate?.let { normalizeToMidnight(Date(it)) } ?: Date(0)
+    val selectedTrip = tripsViewModel.selectedTrip.value!!
```

Comment:
Here instead of asserting it's non null, wdyt of adding a more specific error like in the Tabs screen:
`   val selectedTrip =
      tripsViewModel.selectedTrip.value
          ?: return Text(text = "No trip selected. Should not happen", color = Color.Red)`
This way it's easier to track down than a NullPointerException

Reply:
Actually, since we call Add Activity within a trip anyway, wdyt of sending the trip as a parameter instead of calling `tripsviewmodel.selectedTrip`?

Reply:
You're right, it's better to take all cases into consideration.

Reply:
I agree, if all goes as expected, the selected trip should never be null, so I'm not strongly against leaving this as is. Up to you. In the future please document assumptions like these.

Reply:
I tend to agree with Ioana on this, I think the way she suggested is a bit more elegant. However, I leave it up to you as well!